packages/chain-adapters/src/tron/TronChainAdapter.ts

```diff
@@ -0,0 +1,426 @@
+
+      const signedTxObject: TronUnsignedTx & { signature: string[] } = {
+        ...txToSign.transaction,
+        signature: [signedTx.signature],
+      }
+
+      return JSON.stringify(signedTxObject)
+    } catch (err) {
+      return ErrorHandler(err, {
+        translation: 'chainAdapters.errors.signTransaction',
+      })
+    }
+  }
+
+  async signAndBroadcastTransaction({
+    senderAddress,
+    receiverAddress,
+    signTxInput,
+  }: SignAndBroadcastTransactionInput<KnownChainIds.TronMainnet>): Promise<string> {
+    try {
+      await Promise.all([
+        assertAddressNotSanctioned(senderAddress),
+        receiverAddress !== CONTRACT_INTERACTION && assertAddressNotSanctioned(receiverAddress),
+      ])
+
+      const signedTx = await this.signTransaction(signTxInput as SignTxInput<TronSignTx>)
+
+      return await this.broadcastTransaction({
+        senderAddress,
+        receiverAddress,
+        hex: signedTx,
+      })
+    } catch (err) {
+      return ErrorHandler(err, {
+        translation: 'chainAdapters.errors.signAndBroadcastTransaction',
+      })
+    }
+  }
+
+  async broadcastTransaction({
+    senderAddress,
+    receiverAddress,
+    hex,
+  }: BroadcastTransactionInput): Promise<string> {
+    try {
+      await Promise.all([
+        assertAddressNotSanctioned(senderAddress),
+        receiverAddress !== CONTRACT_INTERACTION && assertAddressNotSanctioned(receiverAddress),
+      ])
+
+      const txHash = await this.providers.http.sendTx({ sendTxBody: { hex } })
+
+      return txHash
+    } catch (err) {
+      return ErrorHandler(err, {
+        translation: 'chainAdapters.errors.broadcastTransaction',
+      })
+    }
+  }
+
+  async getFeeData(
+    _input: GetFeeDataInput<KnownChainIds.TronMainnet>,
+  ): Promise<FeeDataEstimate<KnownChainIds.TronMainnet>> {
+    try {
+      const { fast, average, slow, estimatedBandwidth } =
+        await this.providers.http.getPriorityFees()
+
+      return {
+        fast: {
+          txFee: fast,
+          chainSpecific: { bandwidth: estimatedBandwidth },
+        },
+        average: {
+          txFee: average,
+          chainSpecific: { bandwidth: estimatedBandwidth },
+        },
+        slow: {
+          txFee: slow,
+          chainSpecific: { bandwidth: estimatedBandwidth },
+        },
+      }
+    } catch (err) {
+      return ErrorHandler(err, {
+        translation: 'chainAdapters.errors.getFeeData',
+      })
+    }
+  }
+
+  validateAddress(address: string): Promise<ValidAddressResult> {
+    try {
+      if (!address.startsWith('T')) {
+        return Promise.resolve({ valid: false, result: ValidAddressResultType.Invalid })
+      }
+
+      if (address.length !== 34) {
+        return Promise.resolve({ valid: false, result: ValidAddressResultType.Invalid })
+      }
+
+      return Promise.resolve({ valid: true, result: ValidAddressResultType.Valid })
+    } catch (err) {
```

Comment:
Not sure how we would hit this, but doesn't hurt. Fine to keep as whichever lib we'll use to parse will probably throw. (and generally never a bad idea to handle exceptions early on!)